internal/host/validator.go

```diff
@@ -1498,51 +1267,73 @@ func (v *validator) shouldValidateDNSResolution(cluster *common.Cluster) (bool,
 	return true, ""
 }
 
+// canDetermineImportedClusterManagedNetworking checks if at-least one of the
+// day-2 cluster hosts has obtained the API-connectivity-check ignition file
+// successfully.
+// This is a pre-requisite before we can call importedClusterHasManagedNetworking
+// to determine whether the cluster has managed networking or not.
+func (v *validator) canDetermineImportedClusterManagedNetworking(cluster *common.Cluster) bool {
+	for _, host := range cluster.Hosts {
+		if host.APIVipConnectivity != "" {
+			var response models.APIVipConnectivityResponse
+			if err := json.Unmarshal([]byte(host.APIVipConnectivity), &response); err != nil {
+				v.log.WithError(err).Warnf("Invalid API connectivity response")
+				continue
+			}
+
+			if !response.IsSuccess {
+				continue
+			}
+
+			return true
+		}
+	}
+
+	return false
+}
+
 func domainNameToResolve(c *validationContext, name string) string {
 	return fmt.Sprintf("%s.%s.%s", name, c.cluster.Name, c.cluster.BaseDNSDomain)
 }
 
-func (v *validator) isAPIDomainNameResolvedCorrectly(c *validationContext) ValidationStatus {
+func (v *validator) isAPIDomainNameResolvedCorrectly(c *validationContext) (ValidationStatus, string) {
+	target := "API load balancer"
 	if c.infraEnv != nil {
-		return ValidationSuccessSuppressOutput
+		return ValidationSuccessSuppressOutput, ""
 	}
 	if shouldValidate, ret := v.shouldValidateDNSResolution(c.cluster); !shouldValidate {
-		return ret
+		apiDomainName := domainNameToResolve(c, constants.APIClusterSubdomain)
+		return ret, printIsDomainNameResolvedCorrectly(c, ret, apiDomainName, target)
 	}
 	apiDomainName := domainNameToResolve(c, constants.APIClusterSubdomain)
-	return checkDomainNameResolution(c, apiDomainName)
+	status := checkDomainNameResolution(c, apiDomainName)
+	return status, printIsDomainNameResolvedCorrectly(c, status, apiDomainName, target)
 }
 
-func (v *validator) printIsAPIDomainNameResolvedCorrectly(c *validationContext, status ValidationStatus) string {
-	apiDomainName := domainNameToResolve(c, constants.APIClusterSubdomain)
-	return printIsDomainNameResolvedCorrectly(c, status, apiDomainName, "API")
-}
-
-func (v *validator) isAPIInternalDomainNameResolvedCorrectly(c *validationContext) ValidationStatus {
+func (v *validator) isAPIInternalDomainNameResolvedCorrectly(c *validationContext) (ValidationStatus, string) {
+	target := "internal API load balancer"
 	if c.infraEnv != nil {
-		return ValidationSuccessSuppressOutput
+		return ValidationSuccessSuppressOutput, ""
 	}
 	if shouldValidate, ret := v.shouldValidateDNSResolution(c.cluster); !shouldValidate {
-		return ret
+		apiDomainName := domainNameToResolve(c, constants.InternalAPIClusterSubdomain)
```

Comment:
I thought I had a comment here about how this could be improved.

There's no reason to call this function twice.

Reply:
ref https://github.com/openshift/assisted-service/pull/4097#discussion_r929195361